src/Storages/MergeTree/IMergeTreeDataPart.h

```diff
@@ -446,6 +446,10 @@ class IMergeTreeDataPart : public std::enable_shared_from_this<IMergeTreeDataPar
     void removeDeleteOnDestroyMarker();
     void removeVersionMetadata();
 
+    /// Found column without specific compression and return codec
+    /// for this column with default parameters.
+    CompressionCodecPtr detectDefaultCompressionCodec() const;
```

Comment:
This change was made to test if this function would be able to determine each column's codec type from the `.bin` file on disk. It appears to only return the primary key's codec type, so it doesn't accomplish what we need.

I would appreciate any pointers if there is existing code that can determine a given part and column's codec type from a `.bin` file. I haven't been able to find that code myself and it seems like something that should exist.